tests/Doctrine/Tests/DBAL/Functional/Platform/PlatformRestrictionsTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Doctrine\Tests\DBAL\Functional\Platform;
+
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\Tests\DbalFunctionalTestCase;
+use function mt_rand;
+use function strlen;
+
+/**
+ * This class holds tests that make sure generated SQL statements respect to platform restrictions
+ * like maximum element name length
+ */
+class PlatformRestrictionsTest extends DbalFunctionalTestCase
+{
+    protected function setUp() : void
+    {
+        parent::setUp();
+    }
+
+    /**
+     * Tests element names that are at the boundary of the identifier length limit.
+     * Ensures generated auto-increment identifier name respects to platform restrictions.
+     */
+    public function testMaxIdentifierLengthLimitWithAutoIncrement() : void
+    {
+        $allowedCharacters = 'abcdefghijklmnopqrstuvwxyz';
+        $platform          = $this->connection->getDatabasePlatform();
+        $tableName         = $this->generateRandomString($allowedCharacters, $platform->getMaxIdentifierLength());
```

Comment:
Why does it have to be a random string? `str_repeat('x', $platform->getMaxIdentifierLength())` should suffice.

Reply:
Changed with `str_repeat`.